src/icp_canister_backend/src/lib.rs

```diff
@@ -151,30 +166,89 @@ fn add_deposit(deposit_id: u64, deposit: types::Deposit, user: Principal, assets
         user_deposits.insert(user, user_deposits_list);
     });
 
-    EPISODES.with(|episodes| {
-        let mut episodes_ref = episodes.borrow_mut();
-        let mut episode = episodes_ref.get(&deposit.episode).unwrap_or(Episode {
-            episode_shares: Nat::from(0u64),
-            assets_staked: Nat::from(0u64),
+    if update_pool_stats {
```

Comment:
Why do you make if for the episode state?
I think you need to add it only for the pool stats